common/types_test.go

```diff
@@ -51,3 +54,34 @@ func TestHashJsonValidation(t *testing.T) {
 		}
 	}
 }
+
+func TestAddressUnmarshalJSON(t *testing.T) {
+	var a Address
+	var tests = []struct {
+		Input     string
+		ShouldErr bool
+		Output    *big.Int
+	}{
+		{"", true, nil},
+		{`""`, true, nil},
+		{`"0x"`, true, nil},
+		{`"0x00"`, true, nil},
+		{`"0xG000000000000000000000000000000000000000"`, true, nil},
+		{`"0x0000000000000000000000000000000000000000"`, false, big.NewInt(0)},
+		{`"0x0000000000000000000000000000000000000010"`, false, big.NewInt(16)},
+	}
+	for i, test := range tests {
+		err := a.UnmarshalJSON([]byte(test.Input))
+		if err != nil && !test.ShouldErr {
+			t.Error(i, "expected no error got ", err)
+		}
+		if err == nil {
+			if test.ShouldErr {
+				t.Error(i, "expected error got no error")
+			}
+			if a.Big().Cmp(test.Output) != 0 {
+				t.Error(i, "expected", test.Output, "got", a.Big())
```

Comment:
Please use the format `"test #%d: address mismatch: have %v, want %v"`. Again it's the stdlib approach to describe the error first ("address mismatch") and then say first what we've got and then what we wanted.